src/main/java/org/sagebionetworks/bridge/validators/PermissionValidator.java

```diff
@@ -0,0 +1,50 @@
+package org.sagebionetworks.bridge.validators;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.sagebionetworks.bridge.validators.Validate.CANNOT_BE_NULL;
+import static org.sagebionetworks.bridge.validators.Validate.CANNOT_BE_NULL_OR_EMPTY;
+
+import org.sagebionetworks.bridge.models.permissions.Permission;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+public class PermissionValidator implements Validator {
+    
+    static final String GUID_FIELD = "guid";
+    static final String APP_ID_FIELD = "appId";
+    static final String USER_ID_FIELD = "userId";
+    static final String ACCESS_LEVEL_FIELD = "accessLevel";
+    static final String ENTITY_TYPE_FIELD = "entityType";
+    static final String ENTITY_ID_FIELD = "entityId";
+    
+    public static final PermissionValidator INSTANCE = new PermissionValidator();
+    
+    @Override
+    public boolean supports(Class<?> clazz) {
+        return Permission.class.isAssignableFrom(clazz);
+    }
+    
+    @Override
+    public void validate(Object target, Errors errors) {
+        Permission permission = (Permission) target;
+        
+        if (isBlank(permission.getGuid())) {
+            errors.rejectValue(GUID_FIELD, CANNOT_BE_NULL_OR_EMPTY);
+        }
```

Comment:
Why doesn't validation fail in the service's create method, since you are nulling out GUID in the controller, and new objects won't actually have a guid?

Reply:
I'm generating and overwriting the GUID field before validating in the service's create method. Would it be better to add an "isNew" check to the validator and make validation the first step in the service call?

Reply:
No I think that's fine, sometimes I set values like this and then check them in the validator...it seems robust in case of future changes somewhere.